baseline/client/tests/jspsych-spatial-orientation.test.js

```diff
@@ -88,6 +94,25 @@ describe("jspsych-spatial-orientation.js plugin", () => {
         const data = jsPsych.data.getLastTrialData().values()[0];
         expect(data.completionReason).toBe("timedout");
     });
+
+    it("records all important parameters", () => {
+        const trial = {
+            type: "spatial-orientation",
+            scene: "scene",
+            centerText: "a",
+            topText: "b",
+            pointerText: "c",
+            targetRadians: 0,
+            mode: "test",
+            endTime: -1,
+        };
+        jsPsych.init({timeline: [trial]});
+        const data = jsPsych.data.getLastTrialData().values()[0];
+        expect(data.center).toBe(trial.centerText);
+        expect(data.facing).toBe(trial.topText);
+        expect(data.target).toBe(trial.pointerText);
+        expect(data.mode).toBe(trial.mode);
+    });
```

Comment:
Shouldn't we check for timeLimit as well?

Reply:
Right! I'll expect that it's less than or equal to 0 for this specific test case, since `"timeLimit"` is computed from within the trial plugin from the `"endTime"` parameter rather than being passed directly so it's hard to say what `"timeLimit"` should be exactly.

Reply:
I think that this would probably let you know what it should be exactly:

```
const nowVal = Date.now();
jest.spyOn(global.Date, 'now').mockImplementationOnce(() > nowVal);
```

...and then you can use nowVal to calculate what endTime will be.